### DISCUSSION TITLE
Steam Login Details

### DISCUSSION THREAD

**Discussion by OPNightMarine** [truckersmp-cli/truckersmp-cli#364] [Q&A] [answered]
First off I wanna thank you for the work you've done for the ability to play ats/ets2 multiplayer on linux. Especially with the truckersmp setup file not working with wine.

Question - With having to login with steam account name and password, do I need too worry about any of my login details being stolen? Or is it the same as normally logging into steam?

It may be obvious from the code, but I've no experience to be able to tell.

Much appreciated,
- OPNightMarine

**Comment by Lucki** [comment, marked as answer]
That part asking for the login credentials is [SteamCMD](https://developer.valvesoftware.com/wiki/SteamCMD), the server version of the Steam client downloaded from Valve's Content Delivery Network (CDN):
https://github.com/truckersmp-cli/truckersmp-cli/blob/0374aa16e4d071374d23bf97273b903686e7cfa3/truckersmp_cli/variables.py#L93-L94

Notice here, how only the username from the command line (`"+login", Args.account,`) is forwarded to SteamCMD. SteamCMD asks by itself for additional info when required:
https://github.com/truckersmp-cli/truckersmp-cli/blob/0374aa16e4d071374d23bf97273b903686e7cfa3/truckersmp_cli/steamcmd.py#L69-L132

If you're (rightfully) hesitating to fill in those credentials in a random terminal window you can login using your normal Steam Client and check the box in the login window to remember your information. SteamCMD will pick those up if available and continue without asking any credentials by itself.

**Reply by OPNightMarine** [reply]
Lucki,

Sweet, thank you for all the info and the quick response. Never knew you could login to steam through the terminal with the server version of steam client, didn't even know there was a server version of the steam client. Makes sense now, quite a relief to hear.

Much appreciated,
- OPNightMarine
